web: Detangling some circular dependencies in Admin and User

kensternberg-authentik opened the pull request:
Admin, User, and Flow should not dependend upon each other, at least not in a circular way.  If Admin and User depend on Flow, that's fine, but Flow should not correspondingly depend upon elements of either; if they have something in common, let's put them in `@goauthentik/common` or find some other smart place to store them.

This commit refactors the intentToLabel and actionToLabel functions into `@goauthentik/common/labels` and converts them to static tables for maintenance purposes.

(I've accidentally deleted the checklist.  The code has been formatted, prechecked, manually inspected and automatically tested, to the extent that that last is possible.  No documentation or labeling changes were made.)

kensternberg-authentik committed:
Web: Detangling some circular dependencies in Admin and User

Admin, User, and Flow should not dependend upon each other, at least
not in a circular way.  If Admin and User depend on Flow, that's
fine, but Flow should not correspondingly depend upon elements of
either; if they have something in common, let's put them in
`@goauthentik/common` or find some other smart place to store them.

This commit refactors the intentToLabel and actionToLabel functions
into `@goauthentik/common/labels` and converts them to static tables
for maintenance purposes.

kensternberg-authentik commented on web/.eslintrc.precommit.json:
```diff
@@ -23,6 +23,7 @@
         "@typescript-eslint/ban-ts-comment": "off",
-        "sonarjs/cognitive-complexity": ["error", 9]
+        "sonarjs/cognitive-complexity": ["error", 9],
+        "sonarjs/no-nested-template-literals": "off"
```
I _hated_ doing this, but six of the files I changed had nested template literals _with_ `msg()` strings embedded, which makes them more or less immobile.  Frustrating as heck.

kensternberg-authentik committed:
web: "Consistency is the hobgoblin of small minds" - Ralph Waldo Emerson

kensternberg-authentik commented on web/src/admin/admin-overview/cards/RecentEventsCard.ts:
```diff
@@ -1,7 +1,8 @@
+import { EventGeo } from "@goauthentik/admin/events/utils";
 import { DEFAULT_CONFIG } from "@goauthentik/common/api/config";
 import { EventWithContext } from "@goauthentik/common/events";
+import { actionToLabel } from "@goauthentik/common/labels";
```
_All_ of the changes are of this: referring to the function in a different location, and changing the casing to indicate it's a _function_ rather than a type or class.

kensternberg-authentik committed:
web: I found these confusing to look at, so I added comments.

BeryJu commented on web/src/common/labels.ts:
```diff
@@ -0,0 +1,58 @@
+]);
+
+export const actionToLabel = (action?: EventActions): string =>
+    eventActionToLabel.get(action) ?? action ?? "";
```
~~we should probably have a more visible fallback like we do for the severity below~~

events can also have a custom action (which always has `custom_` prefixed), so as a fallback we should probably just return the action

kensternberg-authentik replied:
That's what this does; only if _that_ fails does it return an empty string. It's the same logic as the original, just less, ah, "iffy."  See: [web/src/admin/events/utils.ts](https://github.com/goauthentik/authentik/pull/6852/files#diff-467f1f91e11f15e16fb5d4465813d492864eddbb23f532b84abc722ab710d8bd)

BeryJu replied:
oh yeah I misread that, LGTM

kensternberg-authentik replied:
Cool, thanks. Yeah, I just started taking the `??` operator seriously recently myself, so I'm still getting used to its semantics. Sorry for going all hypergraphia on ya, I just didn't want to be misunderstood.

kensternberg-authentik committed:
Merge branch 'main' into web/detangling-circular-dependencies

* main: (28 commits)
  root: add option to disable beat when running worker (#6849)
  core: bump ruff from 0.0.287 to 0.0.288 (#6858)
  core: bump goauthentik.io/api/v3 from 3.2023082.6 to 3.2023083.1 (#6859)
  root: fix gunicorn not starting
  providers/saml: set WantAuthnRequestsSigned in metadata (#6851)
  root: fix dumb-init
  web: bump the eslint group in /web with 1 update (#6846)
  web: bump @typescript-eslint/parser from 6.6.0 to 6.7.0 in /web (#6847)
  web: bump API Client version (#6850)
  release: 2023.8.3
  website/docs: prepare 2023.8.3 release notes (#6843)
  core: remove celery's duplicate max_tasks_per_child (#6840)
  providers/scim: check that a provider exists before starting scim task (#6841)
  web/admin: fix flow-search not being able to unset (#6838)
  sources/ldap: dont prefetch useless items (#6812)
  sources/ldap: fix task timeout for ldap_sync_all and ldap_sync_single (#6809)
  root: lock node to 20.5 (#6776)
  web/admin: fix not being able to unset certificates (#6767)
  policies/reputation: require either check to be enabled (#6764)
  stages/password: fix failed_attempts_before_cancel allowing one too m… (#6763)
  ...

kensternberg-authentik committed:
web: remove admin-to-user component reference(s) (#6856)

There was only one: AppIcon.  This has been moved to `components`.

Touching the LibraryApplications page triggered a cyclomatic
complexity check.  Extracting the expansion block and streamlining
the class and style declarations with lit directives helped.

kensternberg-authentik committed:
Merge branch 'main' into web/detangling-circular-dependencies

* main:
  sources/ldap: fix inverted interpretation of FreeIPA nsaccountlock (#6877)
  root: add docker recipe to Makefile
  root: include generated ts client in dev builds
  web: bump the sentry group in /web with 2 updates (#6878)
  core: bump debugpy from 1.7.0 to 1.8.0 (#6867)
  web: bump @lit/localize-tools from 0.6.9 to 0.6.10 in /web (#6869)
  web: bump pyright from 1.1.326 to 1.1.327 in /web (#6870)
  web: bump @lit-labs/context from 0.4.0 to 0.4.1 in /web (#6871)
  ci: bump docker/setup-buildx-action from 2 to 3 (#6872)
  ci: bump docker/login-action from 2 to 3 (#6874)
  ci: bump docker/build-push-action from 4 to 5 (#6875)
  ci: bump docker/setup-qemu-action from 2.2.0 to 3.0.0 (#6873)
  core: bump ruff from 0.0.288 to 0.0.289 (#6868)
  website/docs: tweak Config page (#6854)
